AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientTests.swift

```diff
@@ -241,6 +241,38 @@ class AWSMobileClientTests: XCTestCase {
         signIn(username: username)
     }
     
+    func testSignInWithListener() {
+        let username = "testUser" + UUID().uuidString
+        let signInListenerWasSuccessful = expectation(description: "signIn listener was successful")
+        let signInWasSuccessful = expectation(description: "signIn was successful")
+        signUpAndVerifyUser(username: username)
+
+        AWSMobileClient.sharedInstance().addUserStateListener(self) { (userState, info) in
+            switch (userState) {
+            case .signedIn:
+                signInListenerWasSuccessful.fulfill()
+                print("Listener user is signed in.")
+            default:
+                print("Listener \(userState)")
```

Comment:
I think one of the conditions you're attempting to fix in the bug fix is to prevent the invocation of the user state listener with invalid interim states. Is it worth asserting that you get exactly one invocation of the callback, with the expected state?